test/modules/ModuleLib/TestResults/EdgeHubRestartMessageResult.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults
+{
+    using System;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
+
+    public class EdgeHubRestartMessageResult : MessageTestResult
+    {
+        public EdgeHubRestartMessageResult(
+            string source,
+            DateTime createdAt,
+            string trackingId,
+            string batchId,
+            string sequenceNumber,
+            DateTime edgeHubRestartedTime,
+            HttpStatusCode edgeHubRestartStatusCode,
```

Comment:
why do you need restart status code? if restart is failed, we should fail the test immediately.

Reply:
When a start fails, it can either throw an exception or peacefully return a status code. 
If something is terribly wrong, the expcetion is thrown, the whole module would just crash. 
If something is wrong but not so wacky, I'd rather have it log and send it to TRC so the TRC can decide -- it may fail only restart but it may recover for the next restart cycle.

Reply:
1. Make sure the restart always success before continue testing.